data/recommendation/notify.js

```diff
@@ -1,68 +1,140 @@
+const yo = require('yo-yo');
+
 class Notify {
   start() {
-    this.setup();
+    this.target = this.createBase();
+    document.body.appendChild(this.target);
+    const methodsToBind = ['handleEndWarning', 'handleEndSignup'];
+    for (let key of methodsToBind) { // eslint-disable-line prefer-const
+      this[key] = this[key].bind(this);
+    }
     self.port.on('data', recs => {
-      this.showAdvice(recs);
+      const date = new Date(recs.BreachDate);
+      const year = date.getFullYear();
+      this.render({
+        warn: true,
+        site: recs.Title,
+        year,
+      });
     });
+    this.hasSync = false;
     self.port.on('syncEnabled', () => {
-      this.removeSyncSignup();
+      this.hasSync = true;
     });
     self.port.on('syncDisabled', () => {
-      this.showSyncSignup();
+      this.hasSync = false;
     });
   }
 
-  prettifyDate(ISODate) {
-    return ISODate.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+  render(options) {
+    const div = this.createBase();
+    if (options.warn) {
+      div.appendChild(this.createWarningBox(options.site, options.year));
+    } else { // !warn
+      if (options.changedPassword) {
+        div.appendChild(this.createPasswordChangedBox());
+      } else { // !changedPassword
+        div.appendChild(this.createNoAccountBox());
+      }
+    }
+    yo.update(this.target, div);
```

Comment:
Having a boolean for each box can get you into some weird states. For example, what is the correct behavior when both `options.warn` and `opions.changedPassword` are true? What if you have another box to add? A better approach would probably be a string that states which box to show, and a fallback case. For example

```js
let box;
if (options.boxType === "warn") {
  box = this.createWarningBox(options.site, options.year);
} else if (options.boxType === "changePassword") {
  box = this.createPasswordChangedBox();
} else if (options.boxType === "noAccount") {
  box = this.createNoAccountBox();
} else {
  throw new Error(`Unrecognized boxType "${options.boxType}"`);
}

yo.update(this.target, yo`<div id="panel">${box}</div>`);
```